File: domains/query-execution/steps/raw-analysis/prepare.py
"""4B-5B: apply semantics and select raw-analysis data path."""

from __future__ import annotations


def run(ctx):
    intent = ctx.get("query_intent", {})
    return StepOutcome.next({
        "raw_context": {
            "intent": intent,
            "data_model": "knowledge-store/data-model.json",
            "selected_source": intent.get("source", "datafinder.openapi.analysis_query"),
        }
    })

File: domains/query-execution/steps/raw-analysis/prepare.ts
/** 4B-5B: apply semantics and select raw-analysis data path. */

import { StepOutcome } from "../../scheduler/scheduler.js";

export function run(ctx: Record<string, any>): StepOutcome {
  const intent = ctx["query_intent"] ?? {};
  return StepOutcome.next({
    raw_context: {
      intent,
      data_model: "knowledge-store/data-model.json",
      selected_source: intent["source"] ?? "datafinder.openapi.analysis_query",
    },
  });
}
